src/select.rs

```diff
@@ -85,51 +87,37 @@ where
         }
         <Self as PromptInteraction<T>>::interact(self)
     }
-
-    /// Enable the filter mode
-    pub fn filter_mode(mut self) -> Self {
-        self.enable_filter_mode = true;
-        self
-    }
 }
 
 impl<T: Clone> PromptInteraction<T> for Select<T> {
     fn on(&mut self, event: &Event) -> State<T> {
         let Event::Key(key) = event;
 
-        match key {
-            Key::ArrowUp => {
-                if self.cursor > 0 {
-                    self.cursor -= 1;
-                }
+        if let Some(filter) = &mut self.filter_mode {
+            if let Some(state) = filter.on(key, &self.items) {
+                return state;
             }
-            Key::ArrowDown => {
-                if self.cursor < self.items.len() - 1 {
-                    self.cursor += 1;
-                }
-            }
-            Key::ArrowRight => {
-                if self.input_filter.is_empty() && self.cursor < self.items.len() - 1 {
-                    self.cursor += 1;
+        }
+
+        let (cursor, items) = if let Some(filter) = &mut self.filter_mode {
+            (&mut filter.cursor, &filter.items)
+        } else {
+            (&mut self.cursor, &self.items)
+        };
+
+        match key {
+            Key::ArrowUp | Key::ArrowLeft => {
+                if *cursor > 0 {
+                    *cursor -= 1;
                 }
             }
-            Key::ArrowLeft => {
-                if self.input_filter.is_empty() && self.cursor > 0 {
-                    self.cursor -= 1;
+            Key::ArrowDown | Key::ArrowRight => {
+                if *cursor < items.len() - 1 {
```

Comment:
Should check if items.len() != 0 otherwise it will panic

Reply:
oh, thanks, I'll check it out today later

Reply:
Fixed